Source/player.h

```diff
@@ -313,6 +313,38 @@ struct PlayerStruct {
 	Uint8 *_pDData;
 	Uint8 *_pBData;
 
+	/**
+	 * @brief Gets the most valuable item out of all the player's items that match the given predicate.
+	 * @param itemPredicate The predicate used to match the items.
+	 * @return The most valuable item out of all the player's items that match the given predicate, or 'nullptr' in case no
+	 * matching items were found.
+	 */
+	template <typename TPredicate>
+	const ItemStruct *GetMostValuableItem(TPredicate itemPredicate) const
+	{
+		auto getMostValuableItem = [](const ItemStruct *begin,
+		                               const ItemStruct *end,
+		                               TPredicate itemPredicate,

```

Comment:
To avoid a copy of the lambda this should be `const TPredicate &itemPredicate`.

But even better is to simply capture it: `[&itemPredicate]`

Reply:
> But even better is to simply capture it: [&itemPredicate]

I was avoiding capturing any variables since you said a lambda with captures was not zero-cost.